---
- name: Error handling play
  hosts: localhost
  connection: local
  vars:
   - gmail_password: !vault |
          $ANSIBLE_VAULT;1.1;AES256
          31333565616262346130303331623363316563336535316565643239636438653830366463366331
          6332326164323566653363306361633739653435633736630a646435613664636230363034343836
          33323562386633303565623732663532336336626130633330643264353737353535623363363066
          3133353463626237320a323831353864636539393935323862373234306534626539373333656531
          34346661376433623662633764336563326562643437346336333862373931626563
  tasks:
  - name: Finding files in /home/rajpandey111
    command: "ls /home/rajpandey111/aws_cud"
    register: aws_clo
    ignore_errors: yes
  - debug:
      msg: "This file is not in the location errors are {{aws_clo}}"
  - name: mail the notifications about failures
    mail:
      host: smtp.gmail.com
      subtype: html
      port: 587
      password: "{{ gmail_password }}"
      to: rajpandey111@yahoo.co.in
      from: raj.pandey2212@gmail.com
      username: raj.pandey2212@gmail.com
      subject: Ansible errors
      body: "Heelo from Ansible {{ aws_clo }}"
    delegate_to: localhost
